### DISCUSSION TITLE
After deletion from db with admin ui can't create the same user

### DISCUSSION THREAD

**Discussion by popovidis** [pocketbase/pocketbase#4105] [Q&A] [answered]
Hi there, I am testing setting up auth for a project and during my registration testing I added a few users with the js-sdk by using `await pb.collection("users").create(data);`. After testing that everything worked I deleted all of the accounts with the admin UI, now despite getting no errors when adding the same accounts they do not appear in the admin UI.

Here is an example of my terminal output when my middleware kicks in, then I console.log the data that is fed to the function `await pb.collection("users").create(data);`, followed by the result from the aforementioned function, and then the final console.log which runs after.

```
[middleware] POST http://localhost:3000/register
{
  email: 'nikolay@***',
  password: '123123123',
  passwordConfirm: '123123123'
}
{
  collectionId: '_pb_users_auth_',
  collectionName: 'users',
  created: '2024-01-07 10:34:45.077Z',
  emailVisibility: false,
  id: '50waovvulev5f43',
  updated: '2024-01-07 10:34:45.077Z',
  username: 'users19713',
  verified: false
}
created user: nikolay@***
```

Originally this successfully added the account and I was able to work with the entry, few hours later I deleted it and tried to add it again the same way with the registration page, I would get the same result from the terminal however the timestamp is the same when I originally created the entry (10:34 am, its 5pm now). So I think the entry is still saved somehow?

I tried to delete the entire pb_data folder and ran into the same problem without even being able to add the account again. I deleted my whole folder where the db executable is and still the same issue. I restarted pc, didn't help. The logs in the admin UI don't give any hints either to what could be happening.

I love the project,
Thank you in advance!

**Comment by ganigeorgiev** [comment]
I don't think it is a PocketBase issue as I'm not able to reproduce it.

Once a record is deleted it is permanently removed from the database, we don't store it anywhere else.

Double check that there is no cache involved (_if you use Next.js note that they have custom `fetch` implementation that comes with cache enabled by default_).

You can rule out this by sending direct request to your PocketBase server via Postman, Insomnia, curl, etc.

If you are still not able to resolve it and think that it is a PocketBase issue, please create a **minimal** reproducible repo and I'll have a more detailed look at it.

**Reply by popovidis** [reply]
Aha you're right, I am using server-side functions and it did cache the response... I deleted the .next folder and could add the entry again. 

```
"use server";

import { pb } from "../../lib/pb";

export const registerUser = async (data) => {
  console.log(data);
  try {
    const result = await pb.collection("users").create(data);
    console.log(result);
    console.log(`created user: ${data.email}`);
    return { success: true, message: "Please check your email." };
...
```

Thank you!

**Comment by popovidis** [comment, marked as answer]
I asked for a solution in the Nextjs Discord turns out that it's pb caching the data. The solution is to add no-store like so:

```
    const result = await pb.collection("users").create(data, { cache: "no-store" });
```

**Reply by ganigeorgiev** [reply]
No, it is not PocketBase. PocketBase doesn't have cache by default for the record web APIs.

The JS SDK under the hood uses whatever global `fetch` is available.

As mentioned earlier, Next.js uses a custom `fetch` implementation that overwrites the native one.
For more details about the Next.js caching, please refer to their documentation - https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#caching-data and more specifically the ["Opting out" section](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#opting-out-of-data-caching).
